Library/SPIClient.m

```diff
@@ -1870,7 +1870,14 @@ - (void)handleTerminalStatusResponse:(SPIMessage *)m {
 }
 
 - (void)handleTerminalConfigurationResponse:(SPIMessage *)m {
-    self.terminalModel = [[[SPITerminalConfigurationResponse alloc] initWithMessage:m] getTerminalModel];
+    
+    SPITerminalConfigurationResponse *response = [[SPITerminalConfigurationResponse alloc] initWithMessage:m];
+    
+    if (_pairUsingEftposAddress && response.isSuccess) {
+        self.serialNumber = [response getSerialNumber];
+    }
+    
+    self.terminalModel = [response getTerminalModel];
```

Comment:
Could I please get you to move the setting of the `terminalModel` into the conditional above in case it fails?